includes/class-convertkit-mm-settings.php

```diff
@@ -0,0 +1,269 @@
+
+	}
+
+	/**
+	 * Returns whether debugging is enabled in the Plugin settings.
+	 *
+	 * @since   1.2.2
+	 *
+	 * @return  bool
+	 */
+	public function debug_enabled() {
+
+		return ( $this->settings['debug'] === 'on' ? true : false );
+
+	}
+
+	/**
+	 * Returns the mapping setting for the given MemberMouse Membership Level ID.
+	 *
+	 * @since   1.2.2
+	 *
+	 * @param   int $id     Membership Level ID.
+	 * @return  string          Setting
+	 */
+	public function get_membership_level_mapping( $id ) {
+
+		return $this->get_mapping( $id, 'level' );
+
+	}
+
+	/**
+	 * Returns the mapping setting for the given MemberMouse Membership Level ID
+	 * when the Level is removed from the User.
+	 *
+	 * @since   1.2.2
+	 *
+	 * @param   int $id     Membership Level ID.
+	 * @return  string          Setting
+	 */
+	public function get_membership_level_cancellation_mapping( $id ) {
+
+		return $this->get_mapping( $id, 'level', true );
+
+	}
+
+	/**
+	 * Returns the mapping setting for the given MemberMouse Product ID.
+	 *
+	 * @since   1.2.2
+	 *
+	 * @param   int $id     Membership Level ID.
+	 * @return  string          Setting
+	 */
+	public function get_product_mapping( $id ) {
+
+		return $this->get_mapping( $id, 'product' );
+
+	}
+
+	/**
+	 * Returns the mapping setting for the given MemberMouse Bundle ID.
+	 *
+	 * @since   1.2.2
+	 *
+	 * @param   int $id     Membership Level ID.
+	 * @return  string          Setting
+	 */
+	public function get_bundle_mapping( $id ) {
+
+		return $this->get_mapping( $id, 'bundle' );
+
+	}
+
+	/**
+	 * Returns the mapping setting for the given MemberMouse Bundle ID
+	 * when the Bundle is removed from the User.
+	 *
+	 * @since   1.2.2
+	 *
+	 * @param   int $id     Bundle ID.
+	 * @return  string          Setting
+	 */
+	public function get_bundle_cancellation_mapping( $id ) {
+
+		return $this->get_mapping( $id, 'bundle', true );
+
+	}
+
+	/**
+	 * Returns the mapping setting for the given MemberMouse resource ID, type
+	 * and whether the mapping is for the 'cancellation'.
+	 *
+	 * @since   1.2.2
+	 *
+	 * @param   int    $id                         Level, Product or Bundle ID.
+	 * @param   string $type                       Mapping type (level,bundle,product).
+	 * @param   bool   $is_cancellation_mapping    If the mapping setting is for the 'cancel' action.
+	 * @return  string                              Setting
+	 */
+	private function get_mapping( $id, $type = 'level', $is_cancellation_mapping = false ) {
+
+		// Build key we're looking for in the array of settings.
+		// Membership levels are stored as `convertkit-mapping-ID`, so don't append the type in this instance.
+		$key = 'convertkit-mapping-' . ( $type !== 'level' ? $type . '-' : '' ) . $id;
+
+		// If requesting a 'cancel' setting, append this to the key.
+		// Products don't support cancellation, so ignore this flag if it's for a product.
+		if ( $is_cancellation_mapping && $type !== 'product' ) {
```

Comment:
Wouldn't this encourage silent failures? If a consumer calls this for a product with is_cancellation_mapping true, it will return the mapping for the product happily.